### PR TITLE
[Form] Add support for the `calendar` option in `DateType`

### LINKED ISSUE #20143: [Form] Add support for the `calendar` option in `DateType`
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#57960
| PR author(s) | @alexandre-daubois
| Merged in    | 7.2

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.2 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix #20143

### CODE REVIEW

File: reference/forms/types/date.rst [hunk 1]
```diff
@@ -155,6 +155,19 @@ values for the year, month and day fields::
...
 .. include:: /reference/forms/types/options/view_timezone.rst.inc
 
+``calendar``
+~~~~~~~~~~~~
+
+**type**: ``\IntlCalendar`` **default**: ``null``
+
+The calendar to use for formatting and parsing the date. The value should be
+an instance of the :phpclass:`IntlCalendar` to use. By default, the Gregorian
+calendar with the application default locale calendar is used.
```
Comment (xabbuh) [inline reference/forms/types/date.rst]:
```suggestion
calendar with the application default locale is used.
```

File: reference/forms/types/date.rst [hunk 2]
```diff
@@ -155,6 +155,19 @@ values for the year, month and day fields::
 
 .. include:: /reference/forms/types/options/view_timezone.rst.inc
 
+``calendar``
+~~~~~~~~~~~~
+
+**type**: ``\IntlCalendar`` **default**: ``null``
```
Comment (xabbuh) [inline reference/forms/types/date.rst:161]:
```suggestion
**type**: ``\IntlCalendar|null`` **default**: ``null``
```

Comment (alexandre-daubois) [reply reference/forms/types/date.rst:161]: In the majority of cases, we don't add `|null` to nullable options of form types. I could only find 3 places where this is done. Is there a guideline for this?

Comment (xabbuh) [reply reference/forms/types/date.rst:161]: then let's keep it as is

**Review by xabbuh — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Alex for contributing yet another nice little option and its docs!

**Pull request merged**